### ISSUE TITLE
CJK query expansion produces English-only terms, degrading recall for Chinese queries

### ISSUE THREAD

**Issue by Seraphitah** [tobi/qmd#454]
## Summary

When using `qmd query` with Chinese (CJK) input, the built-in query expansion model (`qmd-query-expansion-1.7B`) expands Chinese queries into English terms, which significantly degrades retrieval quality for Chinese-language document collections.

## Steps to Reproduce

1. Index a collection of Chinese Markdown files
2. Run a Chinese query:
   ```bash
   qmd query "Telegram 配置信息"
   ```
3. Observe that the expanded query contains English terms like "access sara's telegram account" instead of preserving the original Chinese semantics

## Expected Behavior

Query expansion should either:
- Preserve the original language of the query (expand Chinese → Chinese)
- Or provide a config option to disable expansion (e.g., `expandQuery: false` in `index.yml`)

## Current Behavior

The 1.7B expansion model appears to be trained primarily on English data. Chinese queries get "translated" into English during expansion, causing the vector search to match against English-like embeddings rather than the actual Chinese content.

## Workaround

Using typed query syntax (`lex:` / `vec:`) bypasses implicit expansion, but this is not available when QMD is called programmatically via MCP or integrated tools (e.g., OpenClaw `memory_search`).

## Suggestion

One or more of:
1. Add a `expandQuery: true|false` option in `index.yml` or per-collection config
2. Add a `--no-expand` flag to `qmd query`
3. Detect input language and skip expansion for non-English queries
4. Train/ship a multilingual expansion model

## Environment

- qmd version: latest (installed via bun)
- Embedding model: bge-m3 (GGUF)
- OS: macOS (arm64)
- Document language: Chinese (Simplified)

**Comment by kechol** [comment]
+1. I ran into this on a Japanese corpus too, and suggestion No.2 (`--no-expand`) matched what I was looking for.

I drafted a small PR (#622) along those lines in case helpful, but happy to defer if a different approach from your list is preferred.

**Comment by botAGI** [comment]
Can confirm the root cause: the expansion model's training data is 100% English, so any non-English query gets "translated" into English boilerplate (`...is an important concept... in software development`) and the output drifts off the format contract.

I tested this at scale for one language (Russian, 450 held-out queries): stock model produced a valid `hyde/lex/vec` contract in **0/450** cases and emitted English boilerplate in ~23%. Retraining the same base (Qwen3-1.7B) with the exact `finetune/` recipe on ~5k native queries fixes it completely (99.8% valid contract, 0% boilerplate) — so the fix doesn't need any code changes, just a per-language model. If it helps anyone hitting this with Russian: [AGmind/qmd-query-expansion-ru](https://huggingface.co/AGmind/qmd-query-expansion-ru) works as a drop-in via `models: generate:` in `index.yml`.

The recipe generalizes: ~5k seed queries in the target language + teacher distillation into the existing train schema. Happy to share details if a multilingual path is of interest for upstream.
